File: bronze/2775/python/main.py

if (__name__ == "__main__"):
    num = int(input())
    for _ in range(num):

        k = int(input())
        n = int(input())
        data = [0] * n

        for i in range(k +1):
            for j in range(n):
                if (i == 0): data[j] = j +1; continue
                for l in range(n - j -1):
                    data[n - j -1] += data[l]
        print(data[n -1])

File: bronze/2775/c++/main.cpp
#include <iostream>

int main()
{

    int num;
    int k, n;

    std::cin >> num;

    for (int _ = 0; _ < num; _++) {

        std::cin >> k >> n;
        
        int data[n] = {0};
        for (int i = 0; i <= k; i++) {

            for (int j = n; j > 0; j--) {
                if (i == 0) {
                    data[j-1] = j;
                    continue;
                }

                for (int l = 0; l < j; l++) {
                    data[j] += data[l];
                }

            }
            
        }
        std::cout << data[n -1] << std::endl;
    }

    return 0;
}